web/react/components/post_list.jsx

```diff
@@ -128,6 +128,21 @@ export default class PostList extends React.Component {
                 this.userHasSeenNew = true;
             }
             this.isUserScroll = true;
+
+            $('.selected-post').removeClass('selected-post');
+
+            $(React.findDOMNode(this.refs.postlistcontent)).children().each(function select() {
+                if ($(this).position().top > 0) {
+                    if (!$('.selected-post')[0]) {
+                        $(this).addClass('selected-post');
+                    }
+                    return;
+                }
+
+                if ($(this).position().top + $(this).height() / 2 > 0) {
+                    $(this).addClass('selected-post');
+                }
+            });
```

Comment:
I'm having trouble understanding the above loop for a few reasons.
1) I think it's trying to mark the topmost visible post with the selected-post class so that we can make sure it's in view later on, but then I'm not sure why we're marking multiple posts. This could probably use some comments explaining what's happening so we can figure out what this does later on.
2) Could we call it something different than selected-post? Assuming I understand what we're doing here, that doesn't really reflect what it's there for. Maybe something like top-visible-post?
3) Does that return do anything since we can't add duplicate classes to a single dom node? I think jquery's each only breaks if you explicitly return false.